state/execution.go

```diff
@@ -354,6 +349,7 @@ func updateValidators(currentSet *types.ValidatorSet, abciUpdates []abci.Validat
 
 // updateState returns a new State updated according to the header and responses.
 func updateState(
+	logger log.Logger,
```

Comment:
Oh - we were probably logging it where we were before to avoid having to pass the logger here. 

Not sure what exactly would be better here - but we do want to do a better job of separating our logging from our logic. 

Maybe the abci responses should be processed into a new form (ie. call all the PB2TM) before calling this? Then the logging can happen before calling this as well and we wouldn't need to pass the logger in. 

Thoughts?

Reply:
Opened https://github.com/tendermint/tendermint/issues/2865 - thanks!